packages/react-vapor/src/components/color-picker/ColorPicker.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import {ChromePicker, ChromePickerProps, ColorResult} from 'react-color';
+import {connect} from 'react-redux';
+import {isString, noop, uniqueId} from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {addInput, changeInputValue, removeInput} from '../input/InputActions';
+import {InputConnected} from '../input/InputConnected';
+import {InputSelectors} from '../input/InputSelectors';
+
+export interface IColorPickerProps extends ChromePickerProps {
+    id: string;
+    defaultColor?: string;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IColorPickerProps) => {
+    const color = InputSelectors.getValue(state, {id: ownProps.id}) || ownProps.defaultColor;
+    return {
+        color,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IColorPickerProps) => ({
+    componentDidMount: (value: string = '', valid = true, disabled = false) => {
+        dispatch(addInput(ownProps.id, value, valid, disabled));
```

Comment:
I don't understand why we add an input here. We already render an `InputConnected` which is supposed to add itself in the store IIRC.